plugins/api-docs/src/components/GrpcApiDefinitionWidget/GrpcApiDefinitionWidget.tsx

```diff
@@ -18,19 +18,31 @@ import React from 'react';
 import { CodeSnippet } from '@backstage/core-components';
 import { useTheme } from '@material-ui/core/styles';
 import { BackstageTheme } from '@backstage/theme';
+import { ApiEntity } from '@backstage/catalog-model';
+import { GRPCSelfGeneratedAPIReference } from 'grpc-docs';
 
 export type GrpcApiDefinitionWidgetProps = {
-  definition: string;
+  entity: ApiEntity;
 };
 
 export const GrpcApiDefinitionWidget = (
   props: GrpcApiDefinitionWidgetProps,
 ) => {
   const theme = useTheme<BackstageTheme>();
+  if (
+    props.entity.metadata.annotations?.['gendocu-com/grpc-docs'] ===
+    'protoc-gen-doc'
```

Comment:
This should perhaps be a different API `type` instead right? The definition is completely different as far as I understand

Reply:
Yes, you are correct, however in this case the API entity would represent a service's API which is technically a grpc API, but got transformed to be able to render a lot more clear and in the API.
The type of an entity spec is rendered on the API catalog page in the table and if we move this to a different type it would mean that the new type will be rendered in the table.

I think you are right and trying to force it to render the correct type in the table probably not a good idea :D I'll move this to a different type!